### PR TITLE
Support explain create function query

### PR DESCRIPTION
**Pull request by taiyang-li**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Support explain create function query
``` sql 
 :)  explain ast create function  mycast AS  (n) -> cast(n as String);
EXPLAIN AST
CREATE FUNCTION mycast AS n -> CAST(n, 'String')

Query id: 2b579104-8e36-490c-a2ea-42722c4fde5e

┌─explain─────────────────────────────────┐
│ CreateFunctionQuery mycast (children 2) │
│  Identifier mycast                      │
│  Function lambda (children 1)           │
│   ExpressionList (children 2)           │
│    Function tuple (children 1)          │
│     ExpressionList (children 1)         │
│      Identifier n                       │
│    Function CAST (children 1)           │
│     ExpressionList (children 2)         │
│      Identifier n                       │
│      Literal 'String'                   │
└─────────────────────────────────────────┘

11 rows in set. Elapsed: 0.007 sec. 

```

### CODE REVIEW

File: src/Interpreters/InterpreterExplainQuery.cpp [hunk 1]
```diff
@@ -25,6 +25,7 @@
 #include <QueryPipeline/printPipeline.h>
 
 #include <Common/JSONBuilder.h>
+#include "Parsers/formatAST.h"
```
[the remark this answers is not in the document]
Comment (tavplubix) [reply src/Interpreters/InterpreterExplainQuery.cpp:28]: Btw, why do we need to include it if there are no changes in `InterpreterExplainQuery.cpp`?

**Review by tavplubix — COMMENTED** [review]
Could you please add a functional test?

File: src/Interpreters/InterpreterExplainQuery.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (taiyang-li) [reply src/Interpreters/InterpreterExplainQuery.cpp:28]: It is added for debug purpose, removed now.

**Comment by taiyang-li** [comment]
> Could you please add a functional test?

Yes, of course

File: src/Parsers/ASTCreateFunctionQuery.h [hunk 2]
```diff
@@ -10,19 +10,21 @@ namespace DB
 class ASTCreateFunctionQuery : public IAST, public ASTQueryWithOnCluster
 {
 public:
-    String function_name;
+    ASTPtr function_name;
```
[the remark this answers is not in the document]
Comment (taiyang-li) [reply src/Parsers/ASTCreateFunctionQuery.h:13]: It is usefull in ASTCreateFunctionQuery::clone.

**Review by tavplubix — APPROVED** [review]

**Pull request merged**